src/mlpack/core/tree/hrectbound_impl.hpp

```diff
@@ -173,18 +173,40 @@ inline ElemType HRectBound<MetricType, ElemType>::MinDistance(
     // Since only one of 'lower' or 'higher' is negative, if we add each's
     // absolute value to itself and then sum those two, our result is the
     // nonnegative half of the equation times two; then we raise to power Power.
-    sum += pow((lower + fabs(lower)) + (higher + fabs(higher)),
-        (ElemType) MetricType::Power);
+    if (MetricType::Power == 1)
+      sum += (lower + std::fabs(lower)) + (higher + std::fabs(higher));
+    else if (MetricType::Power == 2)
+    {
+      ElemType dist = (lower + std::fabs(lower)) + (higher + std::fabs(higher));
+      sum += dist * dist;
+    }
+    else
+    {
+      sum += pow((lower + fabs(lower)) + (higher + fabs(higher)),
+          (ElemType) MetricType::Power);
+    }
   }
 
   // Now take the Power'th root (but make sure our result is squared if it needs
   // to be); then cancel out the constant of 2 (which may have been squared now)
   // that was introduced earlier.  The compiler should optimize out the if
   // statement entirely.
-  if (MetricType::TakeRoot)
-    return (ElemType) pow((double) sum, 1.0 / (double) MetricType::Power) / 2.0;
+  if (MetricType::Power == 1)
+    return sum * 0.5;
+  else if (MetricType::Power == 2)
+  {
+    if (MetricType::TakeRoot)
+      return std::sqrt(sum) * 0.5;
```

Comment:
I believe the casts to double were done because `std::pow(double, int)` was not available on all systems. It looks like an overload of `std::sqrt()` is available that takes any integral type as input, but returns `double`, so you would have to cast the result back to `ElemType`. It appears that the same comments now apply about `std::pow` since C++11.